### PR TITLE
Fixes #556 - Add Fields/Constants to VB Nav Bar

### LINKED ISSUE #556: VB Nav Bar: Fields (and constants) are not displayed.
Ported from TFS WorkItem: **1036382**

------------------------------

**Repro Steps:**
```
Class C
    Public F As Integer
End Class
```
RESULT: F is not listed in the members drop down
EXPECTED: It should be.

NOTE: The assumption is that old vb assumed all fields were at the top and you would use (Declarations) to get there.

### PR DESCRIPTION
**Pull request by dpen2000**
Fixes #556
Hope this is OK.

### CODE REVIEW

**Comment by davkean** [comment]
@dotnet-bot test eta please
@dotnet-bot test vsi please

**Comment by davkean** [comment]
tag @dotnet/roslyn-ide

**Comment by davkean** [comment]
tag @AnthonyDGreen

File: src/EditorFeatures/VisualBasic/NavigationBar/VisualBasicNavigationBarItemService.vb [hunk 1]
```diff
@@ -269,7 +269,7 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.NavigationBar
                 Return Not p.IsWithEvents
             End If
 
-            Return symbol.Kind = SymbolKind.Event
+            Return symbol.Kind = SymbolKind.Event Or (symbol.Kind = SymbolKind.Field And Not symbol.IsImplicitlyDeclared)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasic/NavigationBar/VisualBasicNavigationBarItemService.vb]: To follow the pattern above, i would change this to:

```vb
if symbol.Kind = SymbolKind.Even
    return True
end if

if symbol.Kind = SymbolKind.Field Andalso Not symbol.IsImplicitlyDeclared
    return true
end if

return false
```

**Comment by CyrusNajmabadi** [comment]
:+1:

File: src/EditorFeatures/VisualBasic/NavigationBar/VisualBasicNavigationBarItemService.vb (hunk 1 above)
Comment (dpen2000) [inline src/EditorFeatures/VisualBasic/NavigationBar/VisualBasicNavigationBarItemService.vb]: @CyrusNajmabadi Thanks! Done this way now.

**Comment by davkean** [comment]
@dotnet-bot test eta please
@dotnet-bot test vsi please

**Comment by jasonmalinowski** [comment]
:+1:. @DustinCampbell might have some hazy recollection of why we didn't do this originally, but it seems like a good idea to me.

@dpen2000: we'll get this merged in a few days. We're shuffling a few branches right now so we can't _quite_ hit merge yet.

**Comment by DustinCampbell** [comment]
The reason was... that it wasn't that way before. It wasn't a good reason.

**Comment by jasonmalinowski** [comment]
And our branch shuffle has completed, so going to merge this. @dpen2000, this change should appear in one of the next previews of Visual Studio "15".

**Pull request merged**